R/check-files.R

```diff
@@ -79,6 +79,12 @@ spell_check_file_text <- function(file, dict){
   spell_check_plain(readLines(file), dict = dict)
 }
 
+spell_check_description_text <- function(file, dict){
+  lines <- readLines(file)
+  lines <- gsub("<http\\S+>", "", lines)
```

Comment:
Don't we still miss the lazy quantifier `+?` to avoid greedy matches until further `>` occurrences? There is also an alternative by excluding `>` alongside spaces in the match.
Maybe it is hard to imagine such a case in a real DESCRIPTION file, here some made-up examples:
``` r
lines <- "<http://u.r.l>(=> pointer) <http://u.r.l>(mizz>speled)"
gsub("<http\\S+>", "", lines)
#> [1] " pointer) speled)"
gsub("<http\\S+?>", "", lines)
#> [1] "(=> pointer) (mizz>speled)"
# alternative to lazy quantifier:
gsub("<http[^\\s>]+>", "", lines)
#> [1] "(=> pointer) (mizz>speled)"
```

Reply:
The `\\S+` will not match spaces so it will only match if the url is directly followed by the `>`.

Reply:
@jeroen, well, my point was rather that `\\S+>` will actually go beyond the `>` at the end of the URL if there is another `>` afterwards (w/o spaces between the two `>`s), as in the examples. As for the alternative `[^\\s>]`, this is a [pretty standard](https://www.regular-expressions.info/repeat.html#lazy) alternative to laziness in similar delimited contexts.

I totally understand though that, in the context of a real DESCRIPTION file, this situation is pretty unlikely, so I guess it is OK to consider this level of accuracy an overkill :smile:. 

On a related note (not sure if this is the right place to ask), wouldn't this type of ad-hoc processing in **spelling** deserve unit-tests? Also given the history and comments in this PR, TDD seams a pretty good fit for this type of things (there is quite a lot done e.g. for [R Markdown](https://github.com/ropensci/spelling/blob/569a24f910d78b9adef6e540e4e0b91849842f94/R/parse-markdown.R#L34)). Apologies if this is inappropriate.